### PR TITLE
feat(gatsby-plugin-utils): add path prefix to image cdn urls

### PR DESCRIPTION
**Pull request by TylerBarnes**
## Description

This PR adds pathPrefix support to Image CDN resolvers.

## Related Issues

https://github.com/gatsbyjs/gatsby/discussions/35147#discussioncomment-3824501
Fixes https://github.com/gatsbyjs/gatsby/pull/36772

## Canaries

 - gatsby-plugin-utils@3.19.0-alpha-image-cdn-pathprefix.48+fa632a40d6
 - gatsby-source-contentful@7.23.0-alpha-image-cdn-pathprefix.29+fa632a40d6
 - gatsby-source-drupal@5.26.0-alpha-image-cdn-pathprefix.48+fa632a40d6
 - gatsby-source-shopify@7.14.0-alpha-image-cdn-pathprefix.48+fa632a40d6
 - gatsby-source-wordpress@6.25.0-alpha-image-cdn-pathprefix.48+fa632a40d6
 - gatsby@4.25.0-alpha-image-cdn-pathprefix.29+fa632a40d6

### CODE REVIEW

File: packages/gatsby-plugin-utils/src/polyfill-remote-file/utils/url-generator.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { URL } from "url"
 import { createContentDigest } from "gatsby-core-utils/create-content-digest"
 import { isImage } from "../types"
 import type { ImageCropFocus, WidthOrHeight } from "../types"
+import { Store } from "gatsby"
```
Comment (wardpeet) [inline packages/gatsby-plugin-utils/src/polyfill-remote-file/utils/url-generator.ts:7]:
```suggestion
import type { Store } from "gatsby"
```

**Review by wardpeet — DISMISSED** [review]
small nit but looks good

**Comment by TylerBarnes** [comment]
Seems like the failing tests are unrelated but I'm not 100% sure. Will look at this again next week.

**Comment by LekoArts** [comment]
@TylerBarnes What's the status here?

**Review by wardpeet — APPROVED** [review]

**Pull request merged**